lib/mailjet/resource.rb

```diff
@@ -131,11 +135,14 @@ def parse_api_json(response_json)
       def create_action_resource_path(id, job_id = nil)
          url_elements = self.resource_path.split("/")
          url_elements.delete_at(url_elements.length-1) if url_elements.last.to_i > 0 #if there is a trailing number for the job id from last call, delete it
-         if self.action != "managemanycontacts" || (self.action == "managemanycontacts" && url_elements[2] == "contactslist")
-           url_elements[3] = id.to_s
-        end
-         url_elements << job_id.to_s if job_id #if job_id exists, ammend it to end of the URI
+
+         if self.action != "managemanycontacts" || (self.action == "managemanycontacts" && url_elements[1] == "contactslist")
```

Comment:
`if self.action != "managemanycontacts" || url_elements[1] == "contactslist"`
because `url_elements[1] == "contactslist"` is evaluated only if `self.action != "managemanycontacts"` is false, so in this case `self.action == "managemanycontacts"` will be true